src/OpenTelemetry.AutoInstrumentation/Configuration/EnvironmentConfigurationTracerHelper.cs

```diff
@@ -56,10 +55,14 @@ public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderB
     public static TracerProviderBuilder AddSdkAspNetInstrumentation(this TracerProviderBuilder builder)
     {
 #if NET462
-        return builder.AddAspNetInstrumentation();
+        builder.AddAspNetInstrumentation();
 #elif NETCOREAPP3_1_OR_GREATER
-        return builder.AddAspNetCoreInstrumentation();
+        // Instrumentation will be added via assembly load detector
+        builder.AddSource("OpenTelemetry.Instrumentation.AspNetCore");
+        builder.AddLegacySource("Microsoft.AspNetCore.Hosting.HttpRequestIn");
```

Comment:
I think we can call this even for .NET Framework as it is possible to use ASP.NET Core on .NET Framework

Reference: https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore#supportedframeworks-body-tab

Reply:
I would move this to the new PR, as it currently replicates the same behaviour we have. Currently we have only one enum member ASP.NET, and based on the framework we either use ASP.NET Core package or ASP.NET framework package.